clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -1830,6 +1830,9 @@ void ItaniumCXXABI::emitVTableDefinitions(CodeGenVTables &CGVT,
   if (VTable->hasInitializer())
     return;
 
+  if (RD->shouldEmitInExternalSource())
+    return;
```

Comment:
> Also, even if the vtable is getting emitted somewhere else, we might want an available_externally vtable.

Out of curiosity, is it helpful to have an available_externally vtable?

Reply:
> This check seems like it's at the wrong level; if we don't want a vtable, why are we calling this in the first place?

Done in the newest update.

Reply:
Yes, and we do emit them in normal builds above -O0 if the vtable is homed elsewhere due to a key function for instance - they can be used to allow for devirtualization

Reply:
Got it. If it is primarily for optimizations, I guess it might not be a blocker. I'll add a FIXME later.

Reply:
But this logic (using available externally definitions) is already built into the modules-codegen work (though it's probably only for functions, I guess it might not be in a place general enough to apply to variables as well - if it either can't be generalized in that way, or even if it just isn't suffiicently general now, I agree that it'd be a separate patch, but probably one worth doing) - again, if/where possible, it'd be nice if these things aligned/involved fewer special cases and inconsistencies.

Reply:
Yes and agreed. I just don't have an idea for how do that.